lib/manifest.go

```diff
@@ -0,0 +1,381 @@
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
+	return a.man.SetUser(user)
+}
+func (a *ACBuild) SetWorkingDir(dir string) (err error) {
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
+	return a.man.SetWorkingDir(dir)
+}
+func (a *ACBuild) SetTag(tag string) (err error) {
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
+	return a.man.SetTag(tag)
+}
+
+func (a *ACBuild) AddDependency(imageName types.ACIdentifier, imageId *types.Hash, labels types.Labels, size uint) (err error) {
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
+	switch m := a.man.(type) {
+	case *appc.Manifest:
+		return m.AddDependency(imageName, imageId, labels, size)
+	}
+	return fmt.Errorf("dependencies only supported in appc builds")
+}
+func (a *ACBuild) RemoveDependency(imageName string) (err error) {
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
+	switch m := a.man.(type) {
+	case *appc.Manifest:
+		return m.RemoveDependency(imageName)
+	}
+	return fmt.Errorf("dependencies only supported in appc builds")
+}
+func (a *ACBuild) AddIsolator(name string, value []byte) (err error) {
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
+	switch m := a.man.(type) {
+	case *appc.Manifest:
+		return m.AddIsolator(name, value)
+	}
+	return fmt.Errorf("dependencies only supported in appc builds")
+}
+func (a *ACBuild) RemoveIsolator(imageName string) (err error) {
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
+	switch m := a.man.(type) {
+	case *appc.Manifest:
+		return m.RemoveIsolator(imageName)
+	}
+	return fmt.Errorf("dependencies only supported in appc builds")
+}
+func (a *ACBuild) SetName(name string) (err error) {
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
+	switch m := a.man.(type) {
+	case *appc.Manifest:
+		return m.SetName(name)
+	}
+	return fmt.Errorf("setting names only supported in appc builds")
+}
+func (a *ACBuild) SetPostStop(exec []string) (err error) {
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
+	switch m := a.man.(type) {
+	case *appc.Manifest:
+		return m.SetPostStop(exec)
+	}
+	return fmt.Errorf("event handlers only supported in appc builds")
+}
+func (a *ACBuild) SetPreStart(exec []string) (err error) {
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
```

Comment:
One option would be to silently drop errors from `a.unlock()`, but I dislike that for obvious reasons